src/ecc_key.c

```diff
@@ -1280,6 +1284,64 @@ int ec_key_sec2prot(struct zpc_ec_key *ec_key, enum ec_key_sec sec)
 	return 0;
 }
 
+int ec_key_clr2prot(struct zpc_ec_key *ec_key, const unsigned char *privkey,
+					unsigned int privlen)
+{
+	struct pkey_kblob2pkey3 io;
+	unsigned char buf[sizeof(struct clearkeytoken) + 80];
+	struct clearkeytoken *clrtok = (struct clearkeytoken *)&buf;
+	int rc;
+
+	memset(buf, 0, sizeof(buf));
+	clrtok->version = 0x02;
+	clrtok->keytype = curve2pkey_keytype[ec_key->curve];
+	switch (clrtok->keytype) {
+	case PKEY_KEYTYPE_ECC_P256:
+	case PKEY_KEYTYPE_ECC_P384:
+	case PKEY_KEYTYPE_ECC_ED25519:
+		memcpy(clrtok->clearkey, privkey, privlen);
+		clrtok->len = privlen;
+		break;
+	case PKEY_KEYTYPE_ECC_P521:
+		memcpy(clrtok->clearkey + 80 - privlen, privkey, privlen);
+		clrtok->len = 80;
+		break;
+	default:
+		memcpy(clrtok->clearkey + 64 - privlen, privkey, privlen);
+		clrtok->len = 64;
+		break;
+	}
+
+	memset(&io, 0, sizeof(io));
+	io.key = buf;
+	switch (clrtok->keytype) {
```

Comment:
you could combine above switch and this one into just one, it has the same cases....

Reply:
Yes indeed. I intended to keep the order of fields to set, but yes, maybe combining the 2 switches is better to read.

Reply:
Ah no ... just saw that the first switch is used to fill out 'struct clearkeytoken' while the second one fills out 'struct pkey_kblob2pkey3'. I would prefer to keep them separate.